update code base to have javadoc tag to explain noinspection tag content

detected at https://github.com/checkstyle/checkstyle/pull/11206#discussion_r797996787

code:
https://github.com/checkstyle/checkstyle/blob/d4228cc1c04f6e18392948986b605cf87738c951/src/main/java/com/puppycrawl/tools/checkstyle/Main.java#L628-L635

has problems that such technical excuse of to suppress IDEA inspection is rendered in final javadoc and can confuse users.
we try to keep reasons of suppressions but in same time we do not want to expose it to users.

proposed solution is too use custom tag for this:
```
$ git diff
diff --git a/pom.xml b/pom.xml
index 31c2853..65fddda 100644
--- a/pom.xml
+++ b/pom.xml
@@ -433,6 +433,10 @@
                 <name>noinspection</name>
                 <placement>X</placement>
               </tag>
+              <tag>
+                <name>noinspectionreason</name>
+                <placement>X</placement>
+              </tag>
             </tags>
           </configuration>
         </plugin>
diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/Main.java b/src/main/java/com/puppycrawl/tools/checkstyle/Main.java
index fec6cbd..f216a44 100644
--- a/src/main/java/com/puppycrawl/tools/checkstyle/Main.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/Main.java
@@ -627,8 +627,10 @@ public final class Main {
 
         /**
          * The checker threads number.
-         * Suppression: CanBeFinal - we use picocli and it use  reflection to manage such fields
-         * This option has been skipped for CLI options intentionally.
+         *
+         * @noinspectionreason CanBeFinal - we use picocli and it use
+         *     reflection to manage such fields
+         *     This option has been skipped for CLI options intentionally.
          *
          * @noinspection CanBeFinal
          */
diff --git a/config/intellij-idea-inspections.xml b/config/intellij-idea-inspections.xml
index de97b29..35a1ab0 100644
--- a/config/intellij-idea-inspections.xml
+++ b/config/intellij-idea-inspections.xml
@@ -2797,7 +2797,7 @@
     <option name="IGNORE_JAVADOC_PERIOD" value="true"/>
     <option name="IGNORE_DUPLICATED_THROWS" value="false"/>
     <option name="IGNORE_POINT_TO_ITSELF" value="false"/>
-    <option name="myAdditionalJavadocTags" value=""/>
+    <option name="myAdditionalJavadocTags" value="noinspectionreason"/>
   </inspection_tool>
   <inspection_tool class="JavaFxColorRgb" enabled="false" level="WARNING"
                    enabled_by_default="false"/>

```

> Yes, any ideas how to make it by existing Checks ?

```
➜  src javac Test.java
➜  src cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="MatchXpath">
            <property name="query" value="//COMMENT_CONTENT[contains(@text, '@noinspection')
                      and not(contains(@text, '@noinspectionreason'))]"/>
            <message key="matchxpath.match"
                     value="'@noinspection' Javadoc tags should be accompanied by a
                     '@noinspectionreason' tag, explaining why we suppressed inspection."/>
        </module>
    </module>
</module>
➜  src cat Test.java
public class Test {

    /**
     * ok
     * @noinspection someInspection
     * @noinspectionreason some reason
     */
    class noInspection1 {

        /**
         * ok
         * @noinspection someInspection
         * @noinspectionreason some reason
         */
        void method1() {

        }

        /**
         * Should be violation here, no noinspectionreason tag
         * @noinspection someInspection
\         */
        void method2() {

        }

        /**
         * ok, no tags at all
         */
        void method3() {

        }
    }

    /**
     * Should be violation here, no noinspectionreason tag
     * @noinspection someInspection
     */
    class noInspection2 {

    }

    /**
     * ok, no tags at all
     */
    class noInspection3 {

    }

}
➜  src java -jar checkstyle-9.2.1-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/nick/IdeaProjects/Tester/src/Test.java:19:11: @noinspection Javadoc tags should be accompanied by a                      @noinspectionreason tag, explaining why we suppressed inspection. [MatchXpath]
[ERROR] /home/nick/IdeaProjects/Tester/src/Test.java:35:7: @noinspection Javadoc tags should be accompanied by a                      @noinspectionreason tag, explaining why we suppressed inspection. [MatchXpath]
Audit done.


```

--- a/.ci/jsoref-spellchecker/whitelist.words
+++ b/.ci/jsoref-spellchecker/whitelist.words
@@ -920,6 +920,7 @@ noenumtrailingcomma
 nofinalizer
 NOI
 noinspection
+noinspectionreason
 nojavadoc
 nolinewrap
 nonascii

--- a/config/intellij-idea-inspections.xml
+++ b/config/intellij-idea-inspections.xml
@@ -2804,7 +2804,7 @@
     <option name="IGNORE_JAVADOC_PERIOD" value="true"/>
     <option name="IGNORE_DUPLICATED_THROWS" value="false"/>
     <option name="IGNORE_POINT_TO_ITSELF" value="false"/>
-    <option name="myAdditionalJavadocTags" value=""/>
+    <option name="myAdditionalJavadocTags" value="noinspectionreason"/>
   </inspection_tool>
   <inspection_tool class="JavaFxColorRgb" enabled="false" level="WARNING"
                    enabled_by_default="false"/>
@@ -3966,7 +3966,10 @@
                         ,com.puppycrawl.tools.checkstyle.gui.ParseTreeTablePresentation,
                         ,com.puppycrawl.tools.checkstyle.gui.TreeTableCellRenderer,
                         ,com.puppycrawl.tools.checkstyle.gui.TreeTableModelAdapter,
-                        ,com.puppycrawl.tools.checkstyle.gui.TreeTable"/>
+                        ,com.puppycrawl.tools.checkstyle.gui.TreeTable,
+                        ,com.puppycrawl.tools.checkstyle.api.Configuration,
+                        ,java.util.EventObject,
+                        ,java.util.Properties,"/>
     <option name="ignoreClassWithoutFields" value="true"/>
   </inspection_tool>
   <inspection_tool class="SerializableInnerClassHasSerialVersionUIDField" enabled="true"

--- a/pom.xml
+++ b/pom.xml
@@ -457,6 +457,10 @@
                 <name>noinspection</name>
                 <placement>X</placement>
               </tag>
+              <tag>
+                <name>noinspectionreason</name>
+                <placement>X</placement>
+              </tag>
             </tags>
           </configuration>
         </plugin>

--- a/src/main/java/com/puppycrawl/tools/checkstyle/DefaultConfiguration.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/DefaultConfiguration.java
@@ -31,8 +31,6 @@ import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
 
 /**
  * Default implementation of the Configuration interface.
- *
- * @noinspection SerializableHasSerializationMethods
  */
 public final class DefaultConfiguration implements Configuration {
 

--- a/src/main/java/com/puppycrawl/tools/checkstyle/ThreadModeSettings.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/ThreadModeSettings.java
@@ -25,6 +25,8 @@ import java.io.Serializable;
  * Thread mode settings for the checkstyle modules.
  *
  * @noinspection SerializableHasSerializationMethods
+ * @noinspectionreason SerializableHasSerializationMethods - we only need serialVersionUID
+ *      to differentiate between threads
  */
 public class ThreadModeSettings implements Serializable {
 

--- a/src/main/java/com/puppycrawl/tools/checkstyle/api/AuditEvent.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/api/AuditEvent.java
@@ -36,7 +36,6 @@ import java.util.EventObject;
  * </p>
  *
  * @see AuditListener
- * @noinspection SerializableHasSerializationMethods
  */
 public final class AuditEvent
     extends EventObject {

--- a/src/main/java/com/puppycrawl/tools/checkstyle/api/Violation.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/api/Violation.java
@@ -44,6 +44,9 @@ import java.util.ResourceBundle.Control;
  * java.text.MessageFormat.
  *
  * @noinspection SerializableHasSerializationMethods, ClassWithTooManyConstructors
+ * @noinspectionreason SerializableHasSerializationMethods - we do not serialize this class
+ * @noinspectionreason ClassWithTooManyConstructors - immutable nature of class requires a
+ *      bunch of constructors
  */
 public final class Violation
     implements Comparable<Violation>, Serializable {
@@ -86,6 +89,10 @@ public final class Violation
      * Arguments for MessageFormat.
      *
      * @noinspection NonSerializableFieldInSerializableClass
+     * @noinspectionreason NonSerializableFieldInSerializableClass - usage of
+     *      'Serializable' for this api class
+     *      is considered as mistake now, but we do not break api without
+     *      good reason
      */
     private final Object[] args;
 

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/OrderedPropertiesCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/OrderedPropertiesCheck.java
@@ -213,7 +213,9 @@ public class OrderedPropertiesCheck extends AbstractFileSetCheck {
     /**
      * Private property implementation that keeps order of properties like in file.
      *
-     * @noinspection ClassExtendsConcreteCollection, SerializableHasSerializationMethods
+     * @noinspection ClassExtendsConcreteCollection
+     * @noinspectionreason ClassExtendsConcreteCollection - we require order from
+     *      file to be maintained by {@code put} method
      */
     private static class SequencedProperties extends Properties {
 

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/UniquePropertiesCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/UniquePropertiesCheck.java
@@ -197,7 +197,9 @@ public class UniquePropertiesCheck extends AbstractFileSetCheck {
     /**
      * Properties subclass to store duplicated property keys in a separate map.
      *
-     * @noinspection ClassExtendsConcreteCollection, SerializableHasSerializationMethods
+     * @noinspection ClassExtendsConcreteCollection
+     * @noinspectionreason ClassExtendsConcreteCollection - we require custom
+     *      {@code put} method to find duplicate keys
      */
     private static class UniqueProperties extends Properties {
 

